mindsdb_native/libs/controllers/transaction.py

```diff
@@ -282,6 +317,48 @@ def run(self):
 
                     output_data[f'{predicted_col}_confidence'][row_number] = probability_true_prediction
 
+            # confidence estimation
+            self.lmd['all_conformal_ranges'] = {}
+            for predicted_col in self.lmd['predict_columns']:
+                X = deepcopy(self.input_data.data_frame)
+                for col in self.lmd['columns_to_ignore'] + self.lmd['predict_columns']:
+                    X.pop(col)
+
+                if not self.lmd['stats_v2'][predicted_col]['typing']['data_type'] == DATA_TYPES.CATEGORICAL:
+                    tol_const = 2  # std devs
+                    tolerance = self.lmd['stats_v2']['train_std_dev'][predicted_col] * tol_const
+                    self.lmd['all_conformal_ranges'][predicted_col] = self.hmd['icp'][predicted_col].predict(X.values)
+
+                    for sample_idx in range(self.lmd['all_conformal_ranges'][predicted_col].shape[0]):
+                        sample = self.lmd['all_conformal_ranges'][predicted_col][sample_idx, :, :]
+                        for idx in range(sample.shape[1]):
+                            significance = (99 - idx) / 100
+                            diff = sample[1, idx] - sample[0, idx]
+                            if diff <= tolerance:
+                                output_data[f'{predicted_col}_confidence'][sample_idx] = significance
+                                output_data[f'{predicted_col}_confidence_range'][sample_idx] = list(sample[:, idx])
+                                break
+                        else:
+                            output_data[f'{predicted_col}_confidence'][sample_idx] = 0.9901  # default
```

Comment:
Setting this to 0.999 makes sense for numerical, since the confidence is given for the range, but it doesn't seem to make sense for categorical, where we want the confidence per a single categroy